ops/charm.py

```diff
@@ -34,30 +34,31 @@
 from ops.framework import EventBase, EventSource, Framework, Object, ObjectEvents
 
 if TYPE_CHECKING:
-    from typing_extensions import Literal, Required, TypedDict
+    from typing import Literal
+
+    from typing_extensions import Required, TypedDict
 
     from ops.framework import Handle, JsonObject, _SerializedData
     from ops.model import Container, Numerical, Relation, Storage
 
     # CharmMeta also needs these.
     _ActionParam = Dict[str, 'JsonObject']  # <JSON Schema definition>
-    _ActionMetaDict = TypedDict(
-        '_ActionMetaDict', {
-            'title': str,
-            'description': str,
-            'params': Dict[str, _ActionParam],
-            'required': List[str]},
-        total=False)
+
+    class _ActionMetaDict(TypedDict, total=False):
+        title: str
+        description: str
+        params: Dict[str, _ActionParam]
+        required: List[str]
 
     _Scopes = Literal['global', 'container']
-    _RelationMetaDict = TypedDict(
-        '_RelationMetaDict', {
-            'interface': Required[str],
-            'limit': int,
-            'scope': _Scopes},
-        total=False)
-
-    _MultipleRange = TypedDict('_MultipleRange', {'range': str})
+
+    class _RelationMetaDict(TypedDict, total=False):
+        interface: Required[str]
+        limit: int
+        scope: _Scopes
+
+    class _MultipleRange(TypedDict):
+        range: str
```

Comment:
This looks weird. Why is _MultipleRange turned into a type, but _StorageMetaDict is not? Also, I think there should be another space here.

Note that I didn't try validating all the changes here, I just happened to see this one.

Reply:
Yeah, that was weird, wasn't it! It turned out it was because some are to define the shape of dicts that have "-" in their keys, like `user-id` and `group-id`, which of course can't be identifiers/class attributes. So those have to remain non-class syntax. I figured it was too confusing and weird to have both types, so I backed this change out.

Which doesn't leave much of a change, but oh well. :-)